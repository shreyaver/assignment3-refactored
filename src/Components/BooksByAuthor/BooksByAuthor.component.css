.booksByAuthor {
    border: 5px solid var(--tan-hide);
    flex: 1;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    min-width: 310px;
    justify-content: space-evenly;
    width: 100%;
    border-radius: 30px;
}
.book:nth-child(even) .bookData{
    background-color: var(--white);
    color: var(--tan-hide);
}
.book:nth-child(odd) .bookData{
    background-color: var(--tan-hide);
    color: var(--blue-dianne);
}
.booksByAuthorContainer {
    margin-top: 10px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center; 
}
.authorName{
    background-color: var(--white);
    color: var(--tan-hide);
    box-sizing: border-box;
    height: 3rem;
    display: flex;
    justify-content: center;
    align-items: center;
    margin-bottom: 5px;
    padding: 0.5rem;
    width: 25%;
    position: relative;
    bottom: -1.5rem;

}
.authorName span {
    flex-grow: 1;
    vertical-align: middle;
    text-align: center;
    font-size: 1.25rem;
    font-style: italic;
}
@media all and (max-width: 600px) {
   .authorName span {
       font-size: 1rem;
   }
   .authorName {
       margin-bottom: 10px;
       width: 35%;
   }
}
