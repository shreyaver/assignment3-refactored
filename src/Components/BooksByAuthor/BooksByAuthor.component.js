import React from 'react';
import PropTypes from 'prop-types';
import Book from '../Book/Book.component';
import './BooksByAuthor.component.css';

const BooksByAuthor = (props) => {
  const booksByAuthor = [];
  const { author, books, handleLikeClick } = props;
  books.forEach((book) => {
    booksByAuthor.push(<Book book={book} key={book.Name} handleLikeClick={handleLikeClick} />);
  });
  return (
    <div className="booksByAuthorContainer">
      <div className="authorName">
        <span>{author}</span>
      </div>
      <div className="booksByAuthor">
        {booksByAuthor}
      </div>
    </div>
  );
};
BooksByAuthor.propTypes = {
  books: PropTypes.arrayOf(
    PropTypes.shape({
      Name: PropTypes.string.isRequired,
      rating: PropTypes.number.isRequired,
      Author: PropTypes.string.isRequired,
      id: PropTypes.number.isRequired,
    }),
  ).isRequired,
  author: PropTypes.string.isRequired,
  handleLikeClick: PropTypes.func.isRequired,
};
export default BooksByAuthor;
