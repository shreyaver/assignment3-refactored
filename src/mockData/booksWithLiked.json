{ "books": {
  "J K Rowling":
    [{ "Author": "J K Rowling",
        "id":70,
        "Name":"Harry Potter and the Deathly Hallows (Harry Potter, #7)",
        "rating":4.62,
        "liked": false
     },
     {"Author": "J K Rowling",
      "id":30,
      "Name":"Harry Potter and the Prisoner of Azkaban (Harry Potter, #3)",
      "rating":4.54,
      "liked": false
     },
     {"Author":"J K Rowling",
      "id":60,
      "Name":"Harry Potter and the Half-Blood Prince (Harry Potter, #6)",
      "rating":4.54,
      "liked": false
    }],
  "Sidney Sheldon":
    [{"Author":"Sidney Sheldon",
      "id":90,
      "Name":"Master of the Game",
      "rating":4.1,
      "liked": false
     },
     {"Author":"Sidney Sheldon",
      "id":80,
      "Name":"If Tomorrow Comes (Tracy Whitney Series, #1)",
      "rating":4.02,
      "liked": false
     },
     {"Author":"Sidney Sheldon",
      "id":100,
      "Name":"Tell Me Your Dreams",
      "rating":3.93,
      "liked": false
     }]
}}