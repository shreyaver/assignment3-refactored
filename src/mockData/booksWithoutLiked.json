{ "J K Rowling":
    [{ "Author": "J K Rowling",
        "id":70,
        "Name":"Harry Potter and the Deathly Hallows (Harry Potter, #7)",
        "rating":4.62
     },
     {"Author": "J K Rowling",
      "id":30,
      "Name":"Harry Potter and the Prisoner of Azkaban (Harry Potter, #3)",
      "rating":4.54
     },
     {"Author":"J K Rowling",
      "id":60,
      "Name":"Harry Potter and the Half-Blood Prince (Harry Potter, #6)",
      "rating":4.54
    }],
  "Sidney Sheldon":
    [{"Author":"Sidney Sheldon",
      "id":90,
      "Name":"Master of the Game",
      "rating":4.1
     },
     {"Author":"Sidney Sheldon",
      "id":80,
      "Name":"If Tomorrow Comes (Tracy Whitney Series, #1)",
      "rating":4.02
     },
     {"Author":"Sidney Sheldon",
      "id":100,
      "Name":"Tell Me Your Dreams",
      "rating":3.93
     }]
}