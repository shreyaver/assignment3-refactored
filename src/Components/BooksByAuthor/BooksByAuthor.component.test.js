import React from 'react';
import renderer from 'react-test-renderer';
import BooksByAuthor from './BooksByAuthor.component';
import mockData from '../../mockData/booksWithLiked.json';

describe('BooksByAuthor', () => {
  it('renders without crashing', () => {
    const booksByAuthor = renderer.create(<BooksByAuthor author={Object.keys(mockData.books)[0]} books={Object.values(mockData.books)[0]} handleLikeClick={jest.fn()} />).toJSON();
    expect(booksByAuthor).toMatchSnapshot();
  });
});
