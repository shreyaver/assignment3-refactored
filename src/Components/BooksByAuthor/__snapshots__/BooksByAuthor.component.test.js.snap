// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooksByAuthor renders without crashing 1`] = `
<div
  className="booksByAuthorContainer"
>
  <div
    className="authorName"
  >
    <span>
      J K Rowling
    </span>
  </div>
  <div
    className="booksByAuthor"
  >
    <div
      className="book"
      style={
        Object {
          "backgroundImage": "url(cover_image.jpeg)",
        }
      }
    >
      <div
        className="imagePanel"
      />
      <div
        className="bookData"
      >
        <p
          className="bookName"
        >
          Harry Potter and the Deathly Hallows (Harry Potter, #7)
        </p>
        <span>
          4.6
          <button
            className="heartIcon"
            type="button"
          />
        </span>
      </div>
    </div>
    <div
      className="book"
      style={
        Object {
          "backgroundImage": "url(cover_image.jpeg)",
        }
      }
    >
      <div
        className="imagePanel"
      />
      <div
        className="bookData"
      >
        <p
          className="bookName"
        >
          Harry Potter and the Prisoner of Azkaban (Harry Potter, #3)
        </p>
        <span>
          4.5
          <button
            className="heartIcon"
            type="button"
          />
        </span>
      </div>
    </div>
    <div
      className="book"
      style={
        Object {
          "backgroundImage": "url(cover_image.jpeg)",
        }
      }
    >
      <div
        className="imagePanel"
      />
      <div
        className="bookData"
      >
        <p
          className="bookName"
        >
          Harry Potter and the Half-Blood Prince (Harry Potter, #6)
        </p>
        <span>
          4.5
          <button
            className="heartIcon"
            type="button"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
